### ISSUE TITLE
2018.2.23 Facebook开源计算机视觉目标检测平台Detectron

### ISSUE THREAD

**Issue by vieyahn2017** [vieyahn2017/IAmInPactera#48]
Facebook开源计算机视觉目标检测平台Detectron

**Comment by vieyahn2017** [comment]
Facebook AI 研究院（FAIR）昨日开源了一款目标检测平台—Detectron，基于Python和Caffe2搭建，其目标是为目标检测研究提供高质量，高性能的代码库。Detectron 包含Mask R-CNN、RetinaNet、Faster R-CNN、RPN、Fast R-CNN以及R-FCN 这些目标检测算法的实现。

Detectron 简介

Detectron 是 FAIR 用于实现最先进的目标检测算法（包括 Mask R-CNN）的软件系统。该系统基于深度学习框架 Caffe 2 ，由 Python 编写而成。

截至营长发稿，目前Detectron在Github上已经获得了5388颗星。
目前，Detectron 已经支持许多研究项目，包括：

Feature Pyramid Networks for Object Detection（https://arxiv.org/abs/1612.03144）
Mask R-CNN（https://arxiv.org/abs/1703.06870）
Detecting and Recognizing Human-Object Interactions（https://arxiv.org/abs/1704.07333）
Focal Loss for Dense Object Detection（https://arxiv.org/abs/1708.02002）
Non-local Neural Networks (https://arxiv.org/abs/1711.07971)
Learning to Segment Every Thing (https://arxiv.org/abs/1711.10370)
Data Distillation: Towards Omni-Supervised Learning (https://arxiv.org/abs/1712.04440)

FAIR 创建 Detectron 是为了向目标检测研究提供高质量、高性能的代码库。它拥有足够的灵活性，可以支持最新研究的快速实施和评估。Detectron 目前囊括了以下对象检测算法的实现：

Mask R-CNN (https://arxiv.org/abs/1703.06870)
RetinaNet (https://arxiv.org/abs/1708.02002)
Faster R-CNN (https://arxiv.org/abs/1506.01497)
RPN (https://arxiv.org/abs/1506.01497)
Fast R-CNN (https://arxiv.org/abs/1504.08083)
R-FCN (https://arxiv.org/abs/1605.06409)

主要使用以下主干网络体系结构：

ResNeXt{50,101,152} (https://arxiv.org/abs/1611.05431)
ResNet{50,101,152} (https://arxiv.org/abs/1512.03385)
Feature Pyramid Networks (https://arxiv.org/abs/1612.03144)
VGG16 (https://arxiv.org/abs/1409.1556)

此外，我们在 Detectron Model Zoo 中提供了大量的基准结果和训练模型以供下载。（https://github.com/facebookresearch/Detectron/blob/master/MODEL_ZOO.md）

如何安装Detectron

详细安装信息，请参考：
https://github.com/facebookresearch/Detectron/blob/master/INSTALL.md

要求：
1.NVIDIA GPU, Linux, Python2
2.Caffe2（请确已将Caffe2升级至支持Detectron模块的版本）和 COCO API

依赖安装
Caffe2安装详见：
https://caffe2.ai/docs/getting-started.html
COCO API：
https://github.com/cocodataset/cocoapi

Detectron安装
Clone the Detectron repository：

# DETECTRON=/path/to/clone/detectron
git clone https://github.com/facebookresearch/detectron $DETECTRON

设置Python模块：

cd $DETECTRON/lib && make

检查Detectron测试通过（比如SpatialNarrowAsOp测试）：

python2 $DETECTRON/tests/test_spatial_narrow_as_op.py

接下来就可以使用Detectron预训练模型进行推理了。

关于安装以及其他问题，请见：
https://github.com/facebookresearch/Detectron/blob/master/INSTALL.md

**Comment by vieyahn2017** [comment]
Docker-开源Detectron

http://blog.csdn.net/pandamax/article/details/79154389

**Comment by vieyahn2017** [comment]
Caffe2 - Detectron 图片测试结果

http://blog.csdn.net/zziahgf/article/details/79148668

**Comment by vieyahn2017** [comment]
http://blog.csdn.net/lanyuxuan100/article/details/79205804

Caffe2 - Detectron 初步使用

**Comment by vieyahn2017** [comment]
Facebook’s AI team Releases Detectron – A Platform for Object Detection Research
We covered Google’s Cloud AutoML Vision last week and, as we predicted, Facebook has already come out with a platform for object detection of it’s own – Detectron.
                                                                          Source: Facebook
Detectron is a software system developed by Facebook’s AI Research team (FAIR) that “implements state-of the art object detection algorithms”. It is written in Python and leverages the Caffee2 deep learning framework underneath.

Detectron aims to provide a high quality and industry standard codebase for object detection research. The results it has posted are incredibly accurate. The image above shows the prediction power of the software. The following object related algorithms are embedded in Detectron:

Mask R-CNN
RetinaNet
Faster R-CNN
RPN
Fast R-CNN
R-FCN
Along with the Python code, FAIR has also released performance baselines for over 70 pre-trained models. Once the model(s) is trained, it can be deployed on the cloud and even on mobile devices.

You can check all of this out on the Github library for Detectron here and the official Facebook launch page Google’s Cloud AutoML Vision.

Our take on this

Detectron’s release will help research communities around the world immeasurably. It’s open source so you can download the code behind this software and even use the plethora of pre-trained models the team has released. From augmented reality to various computer vision tasks, Detectron has a wide variety of uses in the research community.
